pkg/client/cache/reflector_test.go

```diff
@@ -393,11 +371,15 @@ func TestReflectorResync(t *testing.T) {
 			return &api.PodList{ListMeta: unversioned.ListMeta{ResourceVersion: "0"}}, nil
 		},
 	}
-	resyncPeriod := time.Minute
+	resyncPeriod := 5 * time.Second
 	r := NewReflector(lw, &api.Pod{}, s, resyncPeriod)
-	r.now = func() time.Time { return currentTime }
 
-	r.ListAndWatch(wait.NeverStop)
+	stopCh := make(chan struct{})
+	go func() {
+		time.Sleep(resyncPeriod)
```

Comment:
This test will be very fragile & flaky.

We should probably do it the other way around - let the second call to WatchFunc() close the stopChannel.